### DISCUSSION TITLE
Using Node Red Embedded to get full rather than delta values

### DISCUSSION THREAD

**Discussion by mcdonaldajr** [SignalK/signalk-node-red#33] [Q&A]
I would like to do some more advanced AIS data processing using node red. I can use the subscribe node to get deltas, but I really need the full vessel data all at once, so I can do some calculations with the data. I'm sure this must be possible, but I can't see an easy way to do it. Can anyone help?

Thanks, Anthony

**Comment by sbender9** [comment]
In a function node, you can do:

```
var app = node.context().global.get('app')
var allVessels = app.getPath('/vessels')

```
